tests/orchestrator/test_workflow_runner.py

```diff
@@ -474,10 +483,62 @@ def test_resume_failed_task(self, workflow_context, thread_executor):
             new_thread_executor.close()
 
         # Wait for it to finish and assert changes.
+        node = workflow_context.model.node.refresh(node)
         assert node.attributes['invocations'].value == task.max_attempts - 1
         assert task.status == task.SUCCESS
         assert wf_runner.execution.status == wf_runner.execution.SUCCEEDED
 
+    def test_resume_failed_task_and_successful_task(self, workflow_context, thread_executor):
```

Comment:
Add one more test which runs a workflow (`retry_failed` = `False`) where the first task fails, the second one depends on the first one, and on resume validate that the first task doesn't run again but the second one does and succeeds.